Tests/DependencyInjection/SonataMediaExtensionTest.php

```diff
@@ -0,0 +1,122 @@
+    {
+        return new ContainerBuilder();
+    }
+
+    protected function getConfigs()
+    {
+        $configs = array(
+            'sonata_media' => array(
+                'db_driver' => 'doctrine_orm',
+                'default_context' => 'default',
+                'contexts' => array(
+                    'default' => array(
+                        'providers' => array(
+                            'sonata.media.provider.image'
+                        ),
+                        'formats' => array(
+                            'default' => array(
+                                'width' => 100,
+                                'quality' => 100,
+                            ),
+                        ),
+                    ),
+                ),
+                'cdn' => array(
+                    'server' => array(
+                        'path' => '/uploads/media'
+                    ),
+                ),
+                'filesystem' => array(
+                    'local' => array(
+                        'directory' => '%kernel.root_dir%/../web/uploads/media'
+                    ),
+                ),
+            ),
+        );
+
+        return $configs;
+    }
+
+    public function setUp()
+    {
+        parent::setUp();
+        $configs = $this->getConfigs();
+        $this->extension = $this->getExtension();
+        $this->container = new ContainerBuilder();
+        $bundles = array(
+            //
+        );
+        $this->container->setParameter('kernel.bundles', $bundles);
+        $this->extension->load($configs, $this->container);
+    }
+
+    public function testAdapters()
+    {
+        $this->assertTrue($this->container->has('sonata.media.adapter.image.gd'));
+        $this->assertTrue($this->container->has('sonata.media.adapter.image.imagick'));
+        $this->assertTrue($this->container->has('sonata.media.adapter.image.gmagick'));
+        $this->assertTrue($this->container->hasAlias('sonata.media.adapter.image.default'));
+        $this->assertEquals('sonata.media.adapter.image.gd', $this->container->getAlias('sonata.media.adapter.image.default'));
+        if (extension_loaded('gd')) {
+            $this->isInstanceOf('Imagine\\Gd\\Imagine', $this->container->get('sonata.media.adapter.image.default'));
+            $this->isInstanceOf('Imagine\\Gd\\Imagine', $this->container->get('sonata.media.adapter.image.gd'));
+        }
+        if (extension_loaded('imagick')) {
+            $this->isInstanceOf('Imagine\\Imagick\\Imagine', $this->container->get('sonata.media.adapter.image.imagick'));
+        }
+        if (extension_loaded('gmagick')) {
+            $this->isInstanceOf('Imagine\\Gmagick\\Imagine', $this->container->get('sonata.media.adapter.image.gmagick'));
+        }
+        if (!extension_loaded('gd') && !extension_loaded('imagick') && extension_loaded('gmagick')) {
+            throw new RuntimeException('install at least one image adapter (gd, imagick or gmagick)');
```

Comment:
At least, I don't think test should fail if no extension loaded, but should be skipped.

Maybe a refactor had to be done here (e.g. on test per extension).